src/url_builder.py

```diff
@@ -0,0 +1,103 @@
+import json
+import requests
+import networkx as nx
+
+from src.graphquery import GraphQuery
+
+
+class URLBuilder:
+    """class that allows to build the url of a query that will be made for an alias of a FHIR resource to a FHIR api
+
+    Attributes:
+        fhir_api_url: The Service Base URL (e.g. http://hapi.fhir.org/baseR4/)
+        main_resource_alias: alias given to a set of fhir resources of a certain type which are the subject of the api query
+        search_query_url: url which will make it possible to recover the resources of the type of main_alias respecting as well as possible the conditions where on itself and on its neighbors.
+    """
+    def __init__(
+        self,
+        fhir_api_url: str,
+        query_graph: type(GraphQuery),
+        main_resource_alias: str
+    ) -> None:
+        """
+        Arguments:
+            fhir_api_url {str} -- The Service Base URL (e.g. http://hapi.fhir.org/baseR4/)
+            query_graph {type(GraphQuery)} -- instance of a GraphQuery object that gives a graphical representation of the global query
+            main_resource_alias {str} -- alias given to a set of fhir resources of a certain type
+        """
+        self.fhir_api_url = fhir_api_url
+        self.main_resource_alias = main_resource_alias
+        self._query_graph = query_graph
+        
+        self._url_params = None
+        self._get_url_params()
+        self.search_query_url = self._compute_url()
+
+    def _compute_url(self) -> str:
+        """Generates the url which will make it possible to recover the resources of the type of main_alias respecting as well as possible the conditions where on itself and on its neighbors.
+
+        Returns:
+            str -- corresponding API request url
+        """
+        # to do verify self.fhir_api_url finish by '/'
+        search_query_url = (
+            f"{self.fhir_api_url}{self._query_graph.resources_alias_info[self.main_resource_alias]['resource_type']}?"
+            f"{self._url_params or ''}&_format=json"
+        )
+        return search_query_url
+
+    def _get_url_params(self):
+        """retrieves the portions of the url that specify search parameters
+        """
+        for ressource_alias in self._query_graph.resources_alias_info.keys():
+            url_temp = None
+            to_resource, reliable = self._chained_params(ressource_alias)
+
+            if reliable:
+                for search_param, values in self._query_graph.resources_alias_info[ressource_alias][
```

Comment:
Create a variable or a simpler accessor, it's too long :D 
> self._query_graph.resources_alias_info[ressource_alias][....

Reply:
done